Assets/Scripts/Gameplay/UI/PostGameUI.cs

```diff
@@ -52,27 +56,29 @@ void Start()
                 m_WaitOnHostMsg.SetActive(true);
             }
 
-            if (m_NetworkGameStateTransform && m_NetworkGameStateTransform.Value &&
-                m_NetworkGameStateTransform.Value.TryGetComponent(out NetworkGameState networkGameState))
-            {
-                SetPostGameUI(networkGameState.NetworkWinState.winState.Value);
-            }
+            m_WinStatePubSub.Subscribe(SetPostGameUI);
         }
 
-        void SetPostGameUI(WinState winState)
+        void SetPostGameUI(WinStateMessage winStateMessage)
         {
-            // Set end message and background color based last game outcome
-            if (winState == WinState.Win)
-            {
-                m_SceneLight.color = m_WinLightColor;
-                m_WinEndMessage.gameObject.SetActive(true);
-                m_LoseGameMessage.gameObject.SetActive(false);
-            }
-            else if (winState == WinState.Loss)
+            m_WinStatePubSub.Unsubscribe(SetPostGameUI);
```

Comment:
That looks like an evil global variable with extra steps...

Reply:
refactored it to a static instance preserved by one state and destroyed by the other